src/libcore/intrinsics.rs

```diff
@@ -1335,6 +1337,29 @@ mod real_intrinsics {
   }
 }
 
+/// Checks whether `ptr` is properly aligned with respect to
+/// `align_of::<T>()`.
+fn is_aligned<T>(ptr: *const T) -> bool {
```

Comment:
Can you make this `is_aligned_and_non_null`? That's what we mostly want, I think.

Reply:
Oh and could you also use this for the existing checks in `from_raw_parts[_mut]`?

Reply:
should I make this function public then? Since I guess you're talking about from_raw_parts in slice

Reply:
Make it `pub(crate)`, I'd say.